Applications/Spire/Include/Spire/Ui/ContextMenu.hpp

```diff
@@ -0,0 +1,106 @@
+#ifndef SPIRE_CONTEXT_MENU_HPP
+#define SPIRE_CONTEXT_MENU_HPP
+#include <QWidget>
+#include "Spire/Spire/ListModel.hpp"
+#include "Spire/Ui/CheckBox.hpp"
+#include "Spire/Ui/Ui.hpp"
+
+namespace Spire {
+
+  /** Represents a pop-up context menu. */
+  class ContextMenu : public QWidget {
+    public:
+
+      /**
+       * Signals that a menu item was submitted.
+       * @param menu The ContextMenu where the submission is triggered.
+       * @param label The label of the submitted the menu item.
+       */
+      using SubmitSignal = Signal<void (const ContextMenu& menu,
+        const QString& label)>;
+
+      /**
+       * The type of function used to do an action.
+       * @param name The name of the action.
```

Comment:
What is the purpose of the name? What use case does it serve?

Reply:
It can be removed.